src/helpers/range.ts

```diff
@@ -194,3 +195,43 @@ export function copyRangeWithNewSheetId(sheetIdFrom: UID, sheetIdTo: UID, range:
 export function createRange(getters: CoreGetters, sheetId: UID, range?: string): Range | undefined {
   return range ? getters.getRangeFromSheetXC(sheetId, range) : undefined;
 }
+
+/**
+ * Spread multiple colrows zone to one row/col zone and add a many new input range as needed.
+ * For example, A1:B4 will become [A1:A4, B1:B4]
+ */
+export function spreadRange(ranges: string[]): string[] {
+  const postProcessedRanges: string[] = [];
+  for (const range of ranges) {
+    const sheetName = range.includes("!") ? range.split("!")[0] + "!" : "";
```

Comment:
Use `splitReference`,  the exclamantion mark may be part of a sheetname, which would break your code